packages/voting-contract/src/state.rs

```diff
@@ -255,6 +255,10 @@ pub fn proposals<'m, P>() -> Map<'m, u64, Proposal<P>> {
     Map::new("proposals")
 }
 
+pub fn text_proposals<'m, P>() -> Map<'m, u64, Proposal<P>> {
```

Comment:
ah, this must be a function because of `P`, right?

Reply:
I wouldn't store the whole proposal. Which would normalise the state and simplify the Map to a const.

```rust
pub struct ProposalInfo {
  pub title: String,
  pub description: String,
}

const TEXT_PROPOSALS: Map<u64, ProposalInfo> = Map::new("text_proposals");
```

Reply:
Makes sense. If the only purpose of this store is to provide a list of all text proposals, is a map still better than `Item<Vec<ProposalInfo>>`? We could include the proposal_id in the `ProposalInfo` then.

Reply:
`Item<Vec<..>>` has a problem of unbounded size growth. If you end up with a few hundred items there, it begins to give performance issues. If someone can force a few thousand items, you may be able to block the contract (hitting max write size)